pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -287,11 +287,14 @@ const (
 
 // VolumeOptions specifies capacity and tags for a volume.
 type VolumeOptions struct {
-	CapacityGB       int
-	Tags             map[string]string
-	PVCName          string
-	VolumeType       string
-	AvailabilityZone string
+	CapacityGB        int
+	Tags              map[string]string
+	PVCName           string
+	VolumeType        string
+	ZonePresent       bool
```

Comment:
The check for `zones == ""` was already present in the `func ZonesToSet`.

I added `func ValidateZone` to check for `zone == ""`.